mlflow/store/local_artifact_repo.py

```diff
@@ -39,6 +39,27 @@ def log_artifacts(self, local_dir, artifact_path=None):
             mkdir(artifact_dir)
         dir_util.copy_tree(src=local_dir, dst=artifact_dir)
 
+    def download_artifacts(self, artifact_path, dst_path=None):
+        """
+        Download an artifact file or directory to a local directory if applicable, and return a
+        local path for it.
+        The caller is responsible for managing the lifecycle of the downloaded artifacts. If
+        dst_path is unspecified, the downloaded artifact may not be mutated in general.
+
+        :param artifact_path: Relative source path to the desired artifacts.
+        :param dst_path: Absolute path of the local filesystem destination directory to which to
+                         download the specified artifacts. This directory must already exist. If
+                         unspecified, the absolute path of the local artifact will be returned.
```

Comment:
I think this is a little confusing because the `:return:` docs indicate that the method always returns an absolute path on the local filesystem. Can we specify that, because the artifact already exists on the local filesystem, the method will return the pre-existing path?

Reply:
Why is this confusing? It does indeed always return an absolute path on the local filesystem containing the artifacts. Maybe it's the `downloaded` word that's confusing?

Reply:
This is confusing because we're expressing this as if it's special case return behavior, but the language doesn't differ substantially from the language described in `:return:`. The difference to me isn't clear.

Reply:
Nevermind, this looks good. Resolved.